pkg/cosign/cosign.go

```diff
@@ -175,13 +197,30 @@ func VerifySignature(ctx context.Context, opts VerifyOpts) error {
 
 	// --- parse key
 	if opts.Key != "" {
-		verifier, err := LoadKey(ctx, opts.Key, opts.SignatureAlgorithm)
+		verifiers, err := LoadVerifiers(ctx, opts.Key, opts.SignatureAlgorithm)
 		if err != nil {
 			return fmt.Errorf("failed to load key: %w", err)
 		}
-		cosignOpts.SigVerifier = verifier
+		opts.Verifiers = append(opts.Verifiers, verifiers...)
+	}
+
+	var errs []error
+	for _, v := range opts.Verifiers {
+		cosignOpts.SigVerifier = v
+		err := verifySignature(ctx, sigs, imgDigestHash, opts, cosignOpts)
+		if err == nil {
+			return nil
+		}
+		errs = append(errs, err)
 	}
 
+	err = cosign.NewVerificationError("failed to find valid signature for %s matching given identity and annotations using %d loaded verifiers/keys", opts.ImageRef.String(), len(opts.Verifiers))
+	err.(*cosign.VerificationError).SetErrorType(cosign.ErrNoMatchingSignaturesType)
+	logrus.Debugf("%s: %v", err, merr.NewErrors(errs...))
```

Comment:
I _think_ you can use the standard library for this now; see https://pkg.go.dev/errors#Join

Reply:
Absolutely!